legate/core/_lib/context.pyx

```diff
@@ -0,0 +1,88 @@
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# distutils: language = c++
+
+import cython
+from libcpp cimport bool
+from libcpp.string cimport string
+
+
+cdef extern from "core/legate_c.h" nogil:
+    ctypedef enum legate_core_variant_t:
+        pass
+
+cdef extern from "core/task/task_info.h" namespace "legate" nogil:
+    cdef cppclass TaskInfo:
+        bool has_variant(int)
```

Comment:
Can the methods of this class (or any of the below classes) raise C++ exceptions? If so, they should be declared with an `except +` in order for those C++ exceptions to be translated into Python exceptions when they occur. For example:

```
bool has_variant(int) except +
```

If not, please feel free to ignore this comment.